### PR TITLE
Update machine learning tasks topic and add learners

### LINKED ISSUE #6546: Documentation for ML.Net 0.3
There may be a need to update the Documentation (Features available) of ML.Net 0.3. Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bd15175c-5b7d-4310-ecf5-32ffc7bc588b
* Version Independent ID: 38adc676-934e-0ff9-c868-527f075c96df
* Content: [Machine learning tasks](https://docs.microsoft.com/en-us/dotnet/machine-learning/resources/tasks)
* Content Source: [docs/machine-learning/resources/tasks.md](https://github.com/dotnet/docs/blob/master/docs/machine-learning/resources/tasks.md)
* Product: **dotnet-ml**
* GitHub Login: @aditidugar
* Microsoft Alias: **johalex**

### PR DESCRIPTION
**Pull request by JRAlexander**
Update machine learning tasks topic and add learners for each task.

Fixes #6546

[Internal Review URL](https://review.docs.microsoft.com/en-us/dotnet/machine-learning/resources/tasks?branch=pr-en-us-9156)

### CODE REVIEW

**Comment by pkulikov** [comment]
@JRAlexander this is great addition! Looks like #5698 can be closed.

File: docs/machine-learning/resources/tasks.md [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 description: Explore the different machine learning tasks supported in ML.NET.
-ms.date: 06/04/2018
-author: aditidugar
+ms.date: 11/20/2018
+author: jralexander
 ---
 # Machine learning tasks
 
-When building a machine learning model, you first need to define what you are hoping to achieve with your data. After, you can pick the right machine learning task for your situation. The following list describes the different machine learning tasks that you can choose from and some common use cases. 
+When building a machine learning model, you first need to define what you are hoping to achieve with your data. After, you can pick the right machine learning task for your situation. The following list describes the different machine learning tasks that you can choose from and some common use cases.
```
Comment (rpetrusha) [inline docs/machine-learning/resources/tasks.md]: nit: After, --> Afterwards,

File: docs/machine-learning/resources/tasks.md [hunk 2]
```diff
@@ -53,9 +104,53 @@ An [unsupervised machine learning](glossary.md#unsupervised-machine-learning) ta
...
+
+The following learners are available for clustering tasks:
+
+* [KMeansPlusPlusTrainer](xref:Microsoft.ML.Trainers.KMeans.KMeansPlusPlusTrainer)
+
+## Anomaly detection
+
+This task creates an anomaly detection model by using Principal Component Analysis (PCA). PCA-Based Anomaly Detection helps you build a model in scenarios where it is easy to obtain training data from one class, such as valid transactions, but difficult to obtain sufficient samples of the targeted anomalies.
+
+Principal Component Analysis, which is frequently abbreviated to PCA, is an established technique in machine learning. PCA is frequently used in exploratory data analysis because it reveals the inner structure of the data and explains the variance in the data. PCA works by analyzing data that contains multiple variables. It looks for correlations among the variables and determines the combination of values that best captures differences in outcomes. These combined feature values are used to create a more compact feature space called the principal components.
```
Comment (rpetrusha) [inline docs/machine-learning/resources/tasks.md]: nit: abbreviated to --> abbreviated as

File: docs/machine-learning/resources/tasks.md [hunk 3]
```diff
@@ -53,9 +104,53 @@ An [unsupervised machine learning](glossary.md#unsupervised-machine-learning) ta
...
+The following learners are available for ranking tasks:
+
+* [FastTreeRankingFastTreeTrainer](xref:Microsoft.ML.Runtime.FastTreeRankingFastTreeTrainer)
+* [FastTreeRankingTrainer](xref:Microsoft.ML.Trainers.FastTree.FastTreeRankingTrainer)
+* [FastTreeTweedieRegressionFastTreeTrainer](xref:Microsoft.ML.Runtime.FastTreeTweedieRegressionFastTreeTrainer)
+* [LightGbmRankingTrainer](xref:Microsoft.ML.Runtime.LightGBM.LightGbmRankingTrainer)
+
+## Recommendation
+
+A recommendation task enables producing a list of recommended products or services. ML.NET uses Matrix factorization (MF), a common approach to recommendations when you have historical product rating data in your catalog. For example, you have historical movie rating data for your users and want to recommend  other movies they are likely to watch next.
```
Comment (rpetrusha) [inline docs/machine-learning/resources/tasks.md]: nit: I think that there's an extra space between "recommend" and "other"

File: docs/machine-learning/resources/tasks.md [hunk 4]
```diff
@@ -45,6 +78,24 @@ A [supervised machine learning](glossary.md#supervised-machine-learning) task th
...
+
+* [FastTreeRegressionTrainer](xref:Microsoft.ML.Trainers.FastTree.FastTreeRegressionTrainer)
+* [FastTreeRegressionFastTreeTrainer](xref:Microsoft.ML.Runtime.FastTreeRegressionFastTreeTrainer)
+* [FastTreeTweedieTrainer](xref:Microsoft.ML.Trainers.FastTree.FastTreeTweedieTrainer)
+* [LightGbmRegressorTrainer](xref:Microsoft.ML.Runtime.LightGBM.LightGbmRegressorTrainer)
+* [LogisticRegression](xref:Microsoft.ML.Runtime.Learners.LogisticRegression)
+* [OlsLinearRegressionTrainer](xref:Microsoft.ML.Trainers.HalLearners.OlsLinearRegressionTrainer)
+* [OnlineGradientDescentTrainer](xref:Microsoft.ML.Trainers.Online.OnlineGradientDescentTrainer)
+* [PoissonRegression](xref:Microsoft.ML.Trainers.PoissonRegression)
+* [RegressionEnsembleTrainer](Microsoft.ML.Runtime.Ensemble.RegressionEnsembleTrainer)
```
Comment (rpetrusha) [inline docs/machine-learning/resources/tasks.md]: The link is broken -- it needs an **xref:**

**Review by rpetrusha — APPROVED** [review]
This looks good, @JRAlexander. There's one broken link, and I've left a couple of nits that you can choose to address or ignore. Then this is ready to merge.

File: docs/machine-learning/resources/tasks.md (hunk 2 above)
Comment (mairaw) [reply docs/machine-learning/resources/tasks.md]: I'd just do Principal Component Analysis (PCA) as indicated in the style guide:
https://docs.microsoft.com/en-us/style-guide/acronyms#if-you-have-to-use-an-acronym-also-spell-out-the-term-for-clarity

In general, include the acronym in parentheses following the spelled-out term.

**Pull request merged**